File: core/src/main/java/com/sequenceiq/cloudbreak/concurrent/GuardedMethod.java
package com.sequenceiq.cloudbreak.concurrent;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.springframework.stereotype.Component;

@Component
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface GuardedMethod {
    String lockPrefix() default "";
}

File: core/src/main/java/com/sequenceiq/cloudbreak/concurrent/GuardedMethod.kt
package com.sequenceiq.cloudbreak.concurrent

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

import org.springframework.stereotype.Component

@Component
@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class GuardedMethod(val lockPrefix: String = "")
